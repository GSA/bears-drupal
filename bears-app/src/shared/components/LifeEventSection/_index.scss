#usagov-bears-app {
  .benefit-form,
  .verify-selections-view {
    background-color: #1a4480;
    margin: 0 16px;
    padding-bottom: 64px;
  }

  .section-wrapper {
    background-color: #fff;
    border: 1px solid white;
    border-radius: 8px;

    button {
      margin-bottom: 18px;
    }

    button.step-back-link {
      color: #979797;
    }
  }

  .section {
    margin: 50px 20px;
  }

  .section-heading {
    background-color: #1a4480;
    color: #fff;
    margin: 0;
    padding: 64px 16px;
  }

  .usa-fieldset {
    padding-bottom: 30px;
  }

  #benefit-section {
    padding-bottom: 30px;
    margin-bottom: 40px;
    border-bottom: 1px solid lightgrey;
  }
}
