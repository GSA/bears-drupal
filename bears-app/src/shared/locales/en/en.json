{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start",
      "list": [
        {
          "notice": "<div>Your information<strong>will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "heading": "Your potential benefits",
    "description": "<div>You must verify your eligibility and visit each agency to apply. <strong>Check with your city, state, or employer for other benefits.</strong></div>",
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    }
  },
  "notEligibleResults": {
    "heading": "Benefits you did not qualify for",
    "description": "<div>According to your answers you <strong>are not</strong> eligible for these benefits.</div>",
    "heading2": "Non-eligible Results",
    "description2": "Check with your state, city, or employer for other benefits."
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  }
}
