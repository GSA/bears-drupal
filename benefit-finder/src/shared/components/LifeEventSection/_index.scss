@use '../../styles/colors/_index.scss' as color;
@use '../../styles/breakpoints/_index.scss' as *;
@use '../../styles/sizes/_index.scss' as *;
@use '../../styles/mixins/_index.scss' as *;

#benefit-finder {
  .benefit-form,
  .verify-selections-view {
    background-color: color.$marine;
    margin: 0 16px;
    padding-bottom: 64px;
  }

  .section-wrapper {
    background-color: color.$white;
    border: 1px solid white;
    border-radius: 8px;

    button {
      margin-bottom: 18px;
    }

    button.step-back-link {
      color: color.$dark-medium-grey !important;
      background-color: color.$white !important;
      width: auto !important;
    }
  }

  .section {
    margin: 50px 20px;
  }

  .section-nav-btn-group {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;

    button,
    .benefit-modal-group a {
      width: $button-max-width;
    }
  }

  .section-heading {
    @include h3;

    background-color: color.$marine;
    color: color.$white;
    margin: 0;
    padding: 64px 16px;
  }

  .usa-fieldset {
    padding-bottom: 30px;
  }

  #benefit-section {
    padding-bottom: 30px;
    margin-bottom: 40px;
    border-bottom: 1px solid color.$base-light-grey;

    h2.benefit-section-heading {
      @include h3;
    }
  }

  .radio-group {
    display: flex;

    .usa-radio__label {
      margin-right: 32px;
    }
  }

  @media (width >= $desktop) {
    .section {
      margin: 60px;
    }

    .radio-group {
      display: inline;
    }
  }
}
