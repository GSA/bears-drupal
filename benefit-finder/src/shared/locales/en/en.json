{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start:",
      "list": [
        {
          "notice": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "alertBanner": {
    "heading": "Error status",
    "description": "All fields marked required must be completed."
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "qualified": {
      "chevron": {
        "heading": "Your Custom List",
        "description": "<div><p><strong>Here is your list of potential Federal Benefits.</strong> Your state, city or employer may offer other benefits. Visit each agency providing the benefit to learn if you are truly eligible and conduct an actual application.</p></div>"
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you meet the criteria to qualify for these benefits:</strong></div>"
    },
    "notQualified": {
      "chevron": {
        "heading": "Benefits You Did Not Qualify For",
        "description": ""
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you did not meet the criteria to qualify for these benefits:</strong></div>"
    },
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    },
    "relativeBenefits": {
      "heading": "Other benefits that might be relevant to you"
    },
    "shareResults": {
      "heading": "Sharing and printing",
      "shareButton": "Share link",
      "emailButton": "Email results",
      "emailSubject": "Results from USAGovâ€™s benefit finder",
      "printButton": "Print results",
      "description": "Copy a link to this page with the criteria you selected above. Make sure to share it only with those you trust as your answers will be visible."
    }
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  },
  "sectionHeadings": {
    "start": "Complete Questions",
    "continue": "Questions Continued",
    "final": "Final Questions"
  },
  "select": {
    "labelSelect": "Select",
    "defaultValue": "Click to Select",
    "dateDefaultValue": "-Select-"
  },
  "date": {
    "alert": "Date should follow the format 01 - January 22 1977",
    "labelDay": "Day",
    "labelMonth": "Month",
    "labelYear": "Year",
    "monthOptions": [
      {
        "label": "01 - January"
      },
      {
        "label": "02 - February"
      },
      {
        "label": "03 - March"
      },
      {
        "label": "04 - April"
      },
      {
        "label": "05 - May"
      },
      {
        "label": "06 - June"
      },
      {
        "label": "07 - July"
      },
      {
        "label": "08 - August"
      },
      {
        "label": "09 - September"
      },
      {
        "label": "10 - October"
      },
      {
        "label": "11 - November"
      },
      {
        "label": "12 - December"
      }
    ]
  }
}
