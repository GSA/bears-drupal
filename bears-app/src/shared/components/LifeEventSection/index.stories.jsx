import LifeEventSection from './index.jsx'

export default {
  component: LifeEventSection,
  tags: ['autodocs'],
}

export const Primary = {}
