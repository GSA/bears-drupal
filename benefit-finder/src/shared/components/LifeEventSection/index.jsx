import { useState, useEffect, useRef } from 'react'
import PropTypes from 'prop-types'
import createMarkup from '../../utils/createMarkup'
import * as apiCalls from '../../api/apiCalls'
import {
  Alert,
  Button,
  Date,
  Fieldset,
  Heading,
  Radio,
  Select,
  StepIndicator,
  Modal,
} from '../index'
import './_index.scss'

/**
 * a functional component that renders a link as a button
 * @component
 * @param {number} step - inherited current step value
 * @param {function} setStep - inherited function to inc/dec step value
 * @param {object} data - inherieted life event step data
 * @param {function} setStepData - inherited function to set index of step data
 * @param {function} setVerifyStep - inherit view handler
 * @param {function} setViewResults - inherited view handler
 * @param {object} ui - inherited ui translations
 * @return {html} returns a semantic html component that displays a form step
 */
const LifeEventSection = ({
  step,
  setStep,
  data,
  handleData,
  setStepData,
  setVerifyStep,
  setViewResults,
  ui,
}) => {
  // state
  const [modal, setModal] = useState(false)
  const [currentData, setCurrentData] = useState(() => data && data[step - 1])
  const [values, setValues] = useState([])

  // desctructure data
  const {
    stepIndicator,
    buttonGroup,
    reviewSelectionModal,
    requiredLabel,
    sectionHeadings,
  } = ui

  /**
   *
   * start alert
   *
   */
  // establish refs
  const alertFieldRef = useRef(null)

  /**
   * a function that handles class state on our collected required fields
   * @function
   * @return {html}
   */
  const handleAlert = () => {
    // remove the display class from the alert
    alertFieldRef.current.classList.remove('display-none')
    // add to all the collected error fields an error class
    values.forEach(field => {
      field.classList.contains('required-field') &&
        field.classList.add('usa-input--error')
    })
    currentData.completed = false
    window.scrollTo(0, 0)
    return false
  }

  /**
   * a function that triggers the modal to a closed state
   * @function
   */
  const handleSuccess = () => {
    // hide alert by adding the display class
    alertFieldRef.current.classList.add('display-none')
    // remove from all the collected error fields the error class
    values.forEach(field => {
      field.classList.remove('usa-input--error')
    })
    currentData.completed = true
    setValues([])
    return true
  }

  /**
   * a function that collect all the required fields in the current step
   * @function
   */
  const getRequiredFields = () => {
    const collectedNodeList = document.querySelectorAll('.required-field')
    setValues(Array.from(collectedNodeList))
  }

  /**
   * a function that checks if all our required fields have values
   * @function
   * @return {func} either success or alert handler
   */
  const handleCheckRequriedFields = () => {
    // collect all the required fields in the current step
    getRequiredFields()
    // check if any of these elements are valid (will add others later)
    const valid = element => {
      return !element.classList.contains('required-field')
    }

    return values.length === 0 || values.every(valid)
      ? handleSuccess()
      : handleAlert()
  }
  /**
   *
   * end alert
   *
   */

  /**
   * a function that updates our step count and set our data index
   * @function
   * @param {number} updateIndex - value which to update step count
   * @return {null} only executes inherited functions
   */
  const handleForwardUpdate = updateIndex => {
    if (handleCheckRequriedFields() === true) {
      // set complete step usa-step-indicator__segment--complete
      setStep(step + updateIndex)
      setStepData(updateIndex)
    }
  }

  /**
   * a function that updates our step count and set our data index
   * @function
   * @param {number} updateIndex - value which to update step count
   * @return {null} only executes inherited functions
   */
  const handleBackUpdate = updateIndex => {
    setStep(step + updateIndex)
  }

  /**
   * a function that updates our current data state
   * @function
   * @return {object} object as state
   */
  const handleUpdateData = () => {
    handleData([...currentData])
    setCurrentData(currentData)
  }

  /**
   * a function that handles the current selected value of our radio
   * and clears validation error if resolved
   * @function
   * @return {object} object as state
   */
  const handleChanged = (event, criteriaKey) => {
    window.history.replaceState({}, '', window.location.pathname)
    apiCalls.PUT.Data(
      criteriaKey,
      currentData,
      setCurrentData,
      event.target.value
    )
  }

  /**
   * a function that handles the current selected value of our radio
   * and clears validation error if resolved
   * @function
   * @return {object} object as state
   */
  const handleDateChanged = (event, criteriaKey) => {
    window.history.replaceState({}, '', window.location.pathname)
    apiCalls.PUT.DataDate(
      criteriaKey,
      currentData,
      setCurrentData,
      event.target.value,
      event.target.id
    )
  }

  // manage the display of our modal initializer
  useEffect(() => {
    data && step === data.length ? setModal(true) : setModal(false)
  }, [currentData, data, modal, step])

  // check for all required fields and scroll to top on mount
  useEffect(() => {
    window.scrollTo(0, 0)
    getRequiredFields()
  }, [])

  console.log('currentData', currentData, 'step', step)

  return (
    data && (
      <>
        <Heading className="section-heading" headingLevel={1}>
          {step === data.length
            ? `${sectionHeadings.final}`
            : step - 1 === 0
            ? `${sectionHeadings.start}`
            : `${sectionHeadings.continue}`}
        </Heading>
        <div className="section-wrapper">
          <div className="section">
            <StepIndicator
              current={step - 1}
              setCurrent={setStep}
              data={data}
              backLinkLabel={stepIndicator.stepBackLink}
              handleCheckRequriedFields={() => handleCheckRequriedFields()}
              completed={currentData.section.completed}
              key={`step-indicator-${sectionHeadings}`}
            />
            {currentData && (
              <div id="benefit-section" onChange={() => handleUpdateData}>
                <Alert
                  alertFieldRef={alertFieldRef}
                  heading={ui.alertBanner.heading}
                  description={ui.alertBanner.description}
                  error
                ></Alert>
                <Heading headingLevel={2}>
                  {currentData.section.heading}
                </Heading>
                <div
                  dangerouslySetInnerHTML={createMarkup(
                    currentData.section.description
                  )}
                ></div>

                {currentData.section.fieldsets.map((item, i) => {
                  const Input = ({ item, children, index, hidden }) =>
                    item.fieldset.inputs[0].inputCriteria.type === 'Select' ? (
                      //
                      //
                      // case select
                      //
                      //
                      <>
                        <Fieldset
                          key={`select-${item.fieldset.criteriaKey}-${index}`}
                          legend={item.fieldset.legend}
                          hint={item.fieldset.hint}
                          required={item.fieldset.required}
                          requiredLabel={requiredLabel}
                          hidden={hidden && hidden}
                        >
                          {item.fieldset.inputs.map((input, index) => {
                            const fieldSetId = `${item.fieldset.criteriaKey}_${index}`
                            const inputValues = input.inputCriteria.values
                            const defaultSelected = inputValues.find(
                              value => value.selected !== undefined
                            )

                            return (
                              <div key={fieldSetId}>
                                <Select
                                  required={
                                    defaultSelected === undefined &&
                                    item.fieldset.required
                                  }
                                  ui={ui?.select}
                                  htmlFor={fieldSetId}
                                  key={fieldSetId}
                                  options={inputValues}
                                  selected={defaultSelected?.value}
                                  onChange={event =>
                                    handleChanged(
                                      event,
                                      item.fieldset.criteriaKey
                                    )
                                  }
                                />
                              </div>
                            )
                          })}
                        </Fieldset>
                        {children || null}
                      </>
                    ) : item.fieldset.inputs[0].inputCriteria.type ===
                      'Radio' ? (
                      //
                      //
                      // case radio
                      //
                      //
                      <>
                        <Fieldset
                          key={`radio-${item.fieldset.criteriaKey}-${index}`}
                          legend={item.fieldset.legend}
                          hint={item.fieldset.hint}
                          required={item.fieldset.required}
                          requiredLabel={requiredLabel}
                          hidden={hidden && hidden}
                        >
                          {item.fieldset.inputs.map((input, index) => {
                            const fieldSetId = `${item.fieldset.criteriaKey}_${index}`

                            const inputValues = input.inputCriteria.values
                            const optionSelected = inputValues.find(
                              value => value.selected !== undefined
                            )

                            return (
                              <div className="radio-group" key={fieldSetId}>
                                {/* map the options */}
                                {input.inputCriteria.values.map(
                                  (option, index) => {
                                    const inputId = `${fieldSetId}_${index}`

                                    return (
                                      <Radio
                                        required={
                                          !optionSelected &&
                                          item.fieldset.required
                                        }
                                        name={fieldSetId}
                                        key={inputId}
                                        id={inputId}
                                        label={option.value}
                                        value={option.value}
                                        checked={option.selected || false}
                                        onChange={event => {
                                          handleChanged(
                                            event,
                                            item.fieldset.criteriaKey
                                          )
                                        }}
                                      />
                                    )
                                  }
                                )}
                              </div>
                            )
                          })}
                        </Fieldset>
                        {children || null}
                      </>
                    ) : item.fieldset.inputs[0].inputCriteria.type ===
                      'Date' ? (
                      //
                      //
                      // case date
                      //
                      //
                      <>
                        <Fieldset
                          key={`date-${item.fieldset.criteriaKey}-${index}`}
                          legend={item.fieldset.legend}
                          hint={item.fieldset.hint}
                          required={item.fieldset.required}
                          requiredLabel={requiredLabel}
                          hidden={hidden && hidden}
                        >
                          {item.fieldset.inputs.map((input, index) => {
                            const fieldSetId = `${item.fieldset.criteriaKey}_${index}`
                            return (
                              <div key={fieldSetId}>
                                <Date
                                  required={
                                    Object.keys(
                                      input.inputCriteria.values[0]?.value
                                    ).length < 3
                                      ? item.fieldset.required
                                      : 'FALSE'
                                  }
                                  value={input.inputCriteria.values[0]?.value}
                                  onChange={event =>
                                    handleDateChanged(
                                      event,
                                      item.fieldset.criteriaKey
                                    )
                                  }
                                  ui={ui}
                                />
                              </div>
                            )
                          })}
                        </Fieldset>
                        {children || null}
                      </>
                    ) : null

                  const parentElement = ({ item, i }) =>
                    Input({ item, index: i })

                  const checkParentValue = item => {
                    const selectedParentValue = apiCalls.GET.SelectedValue(item)

                    const hidden =
                      selectedParentValue?.value !==
                      item.fieldset.inputs[0].inputCriteria
                        .childDependencyOption

                    return hidden
                  }

                  const parentWithChildElement = ({ item, i }) => {
                    return Input({
                      item,
                      index: i,
                      children: item.fieldset.children.map(
                        (child, i) =>
                          child.fieldsets.length &&
                          child.fieldsets.map((childItem, childItemIndex) =>
                            Input({
                              item: childItem,
                              index: childItemIndex,
                              hidden: checkParentValue(item),
                            })
                          )
                      ),
                    })
                  }

                  // children check
                  const fieldSet = () =>
                    item.fieldset.children.length > 0
                      ? parentWithChildElement({ item, i })
                      : parentElement({ item, i })

                  return fieldSet()
                })}
              </div>
            )}
            <Button secondary onClick={() => handleBackUpdate(-1)}>
              {buttonGroup[0].value}
            </Button>
            {modal === false ? (
              <Button onClick={() => handleForwardUpdate(1)}>
                {buttonGroup[1].value}
              </Button>
            ) : (
              <div>
                <Modal
                  id="nav-modal"
                  modalHeading={reviewSelectionModal.heading}
                  navItemOneLabel={reviewSelectionModal.buttonGroup[0].value}
                  navItemOneFunction={setVerifyStep}
                  navItemTwoLabel={reviewSelectionModal.buttonGroup[1].value}
                  navItemTwoFunction={setViewResults}
                  triggerLabel={buttonGroup[1].value}
                  handleCheckRequriedFields={handleCheckRequriedFields}
                />
              </div>
            )}
          </div>
        </div>
      </>
    )
  )
}

LifeEventSection.propTypes = {
  props: PropTypes.any,
  step: PropTypes.number,
  setStep: PropTypes.func,
  data: PropTypes.array,
  setStepData: PropTypes.func,
  setVerifyStep: PropTypes.func,
  setViewResults: PropTypes.func,
  ui: PropTypes.object,
}

export default LifeEventSection
