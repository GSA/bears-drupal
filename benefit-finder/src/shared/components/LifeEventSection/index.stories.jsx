import LifeEventSection from './index.jsx'
import content from '../../api/mock-data/current.js'
import * as en from '../../locales/en/en.json'

const { data } = JSON.parse(content)
const { lifeEventForm } = data

export default {
  component: LifeEventSection,
  tags: ['autodocs'],
  args: {
    ui: { ...en },
    data: lifeEventForm.sectionsEligibilityCriteria,
    step: 1,
  },
}

export const Primary = {}
