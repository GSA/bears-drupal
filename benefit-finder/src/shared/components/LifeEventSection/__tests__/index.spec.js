import renderer from 'react-test-renderer'
import LifeEventSection from '../index.jsx'
import content from '../../../api/mock-data/content-data.js'
import * as en from '../../../locales/en/en.json'

const { lifeEventForm } = JSON.parse(content)

describe('LifeEventSection', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(
      <LifeEventSection
        ui={en}
        data={lifeEventForm.sectionsEligibilityCriteria}
        step={1}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
