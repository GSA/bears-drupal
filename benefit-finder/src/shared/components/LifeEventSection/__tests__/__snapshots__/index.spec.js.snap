// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifeEventSection renders a match to the previous snapshot 1`] = `
Array [
  <h1
    className=""
    id="skip-to-h1"
  >
    Complete Questions
  </h1>,
  <div
    className="section-wrapper"
  >
    <div
      className="section"
    >
      <div>
        <div
          aria-label="progress"
          className="usa-step-indicator"
        >
          <ol
            className="usa-step-indicator__segments"
          >
            <li
              className="usa-step-indicator__segment usa-step-indicator__segment--current usa-step-indicator__segment--complete"
              onClick={[Function]}
            >
              <span
                className="usa-step-indicator__segment-label"
              >
                About the applicant
                 
                <span
                  className="usa-sr-only"
                />
              </span>
            </li>
            <li
              className="usa-step-indicator__segment usa-step-indicator__segment "
              onClick={[Function]}
            >
              <span
                className="usa-step-indicator__segment-label"
              >
                About the deceased
                 
                <span
                  className="usa-sr-only"
                />
              </span>
            </li>
          </ol>
        </div>
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
      </div>
      <div
        id="benefit-section"
        onChange={[Function]}
      >
        <div
          className=""
          role="alert"
        >
          <div
            className="usa-alert__body"
          >
            <h4
              className="usa-alert__heading"
            >
              Error status
            </h4>
            <p
              className="usa-alert__text"
            >
              All fields marked required must be completed.
            </p>
          </div>
        </div>
        <h2
          className=""
          id=""
        >
          About the applicant
        </h2>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>About the person looking for benefits</p>",
            }
          }
        />
        <fieldset
          className=""
        >
          Applicant's DOB: 1969/08/15
          <legend
            className="usa-legend legend"
          >
            Date of birth:
            <span
              className="usa-hint usa-hint--required required"
              title="required"
            >
              (Required)
            </span>
          </legend>
          <div>
            <div
              className="usa-memorable-date"
            >
              <div
                className="usa-form-group usa-form-group--month usa-form-group--select"
              >
                <label
                  className="usa-label"
                  htmlFor="date_of_birth_month"
                >
                  Month
                </label>
                <select
                  aria-describedby="mdHint"
                  className="usa-select required-field"
                  id="date_of_birth_month"
                  name="date_of_birth_month"
                  onChange={[Function]}
                  required="required"
                  value=""
                >
                  <option
                    disabled={true}
                    value=""
                  >
                    -Select-
                  </option>
                  <option
                    value={0}
                  >
                    01 - January
                  </option>
                  <option
                    value={1}
                  >
                    02 - February
                  </option>
                  <option
                    value={2}
                  >
                    03 - March
                  </option>
                  <option
                    value={3}
                  >
                    04 - April
                  </option>
                  <option
                    value={4}
                  >
                    05 - May
                  </option>
                  <option
                    value={5}
                  >
                    06 - June
                  </option>
                  <option
                    value={6}
                  >
                    07 - July
                  </option>
                  <option
                    value={7}
                  >
                    08 - August
                  </option>
                  <option
                    value={8}
                  >
                    09 - September
                  </option>
                  <option
                    value={9}
                  >
                    10 - October
                  </option>
                  <option
                    value={10}
                  >
                    11 - November
                  </option>
                  <option
                    value={11}
                  >
                    12 - December
                  </option>
                </select>
              </div>
              <div
                className="usa-form-group usa-form-group--day"
              >
                <label
                  className="usa-label"
                  htmlFor="date_of_birth_day"
                >
                  Day
                </label>
                <input
                  aria-describedby="mdHint"
                  className="usa-input required-field"
                  id="date_of_birth_day"
                  inputMode="numeric"
                  name="date_of_birth_day"
                  onChange={[Function]}
                  value=""
                />
              </div>
              <div
                className="usa-form-group usa-form-group--year"
              >
                <label
                  className="usa-label"
                  htmlFor="date_of_birth_year"
                >
                  Year
                </label>
                <input
                  aria-describedby="mdHint"
                  className="usa-input required-field"
                  id="date_of_birth_year"
                  inputMode="numeric"
                  name="date_of_birth_year"
                  onChange={[Function]}
                  value=""
                />
              </div>
            </div>
          </div>
        </fieldset>
        <fieldset
          className=""
        >
          <legend
            className="usa-legend legend"
          >
            Applicantâ€™s relationship to the deceased:
          </legend>
          <div>
            <label
              className=""
              htmlFor="applicant_relationship_to_the_deceased_0"
            >
              Select
            </label>
            <select
              className=""
              id="applicant_relationship_to_the_deceased_0"
              name="applicant_relationship_to_the_deceased_0"
              onChange={[Function]}
              value=""
            >
              <option
                disabled={true}
                value=""
              >
                Click to Select
              </option>
              <option
                value="Spouse"
              >
                Spouse
              </option>
              <option
                value="Child"
              >
                Child
              </option>
              <option
                value="Parent"
              >
                Parent
              </option>
              <option
                value="Other family member"
              >
                Other family member
              </option>
              <option
                value="Personal or official representative"
              >
                Personal or official representative
              </option>
            </select>
          </div>
        </fieldset>
        <fieldset
          className=""
        >
          <legend
            className="usa-legend legend"
          >
            Marital Status
          </legend>
          <div>
            <label
              className=""
              htmlFor="applicant_marital_status_0"
            >
              Select
            </label>
            <select
              className=""
              id="applicant_marital_status_0"
              name="applicant_marital_status_0"
              onChange={[Function]}
              value=""
            >
              <option
                disabled={true}
                value=""
              >
                Click to Select
              </option>
              <option
                value="Married"
              >
                Married
              </option>
              <option
                value="Unmarried"
              >
                Unmarried
              </option>
              <option
                value="Widowed"
              >
                Widowed
              </option>
              <option
                value="Divorced"
              >
                Divorced
              </option>
            </select>
          </div>
        </fieldset>
        <fieldset
          className=""
        >
          <legend
            className="usa-legend legend"
          >
            Are you a U.S. citizen or eligible non-citizen?
          </legend>
          <div
            className="radio-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_citizen_status_0_0"
                name="applicant_citizen_status_0"
                onChange={[Function]}
                type="radio"
                value="Yes"
              />
              <label
                className=""
                htmlFor="applicant_citizen_status_0_0"
              >
                Yes
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_citizen_status_0_1"
                name="applicant_citizen_status_0"
                onChange={[Function]}
                type="radio"
                value="No"
              />
              <label
                className=""
                htmlFor="applicant_citizen_status_0_1"
              >
                No
              </label>
            </div>
          </div>
        </fieldset>
        <fieldset
          className=""
        >
          <legend
            className="usa-legend legend"
          >
            Are you caring for a child of the deceased who is disabled or under the age of 16?
          </legend>
          <div
            className="radio-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_care_for_child_0_0"
                name="applicant_care_for_child_0"
                onChange={[Function]}
                type="radio"
                value="Yes"
              />
              <label
                className=""
                htmlFor="applicant_care_for_child_0_0"
              >
                Yes
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_care_for_child_0_1"
                name="applicant_care_for_child_0"
                onChange={[Function]}
                type="radio"
                value="No"
              />
              <label
                className=""
                htmlFor="applicant_care_for_child_0_1"
              >
                No
              </label>
            </div>
          </div>
        </fieldset>
        <fieldset
          className=""
        >
          <legend
            className="usa-legend legend"
          >
            Did you pay for funeral or burial expenses?
          </legend>
          <div
            className="radio-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_paid_funeral_expenses_0_0"
                name="applicant_paid_funeral_expenses_0"
                onChange={[Function]}
                type="radio"
                value="Yes"
              />
              <label
                className=""
                htmlFor="applicant_paid_funeral_expenses_0_0"
              >
                Yes
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_paid_funeral_expenses_0_1"
                name="applicant_paid_funeral_expenses_0"
                onChange={[Function]}
                type="radio"
                value="No"
              />
              <label
                className=""
                htmlFor="applicant_paid_funeral_expenses_0_1"
              >
                No
              </label>
            </div>
          </div>
        </fieldset>
        <fieldset
          className=""
        >
          <legend
            className="usa-legend legend"
          >
            Did you pay for funeral or burial expenses?
          </legend>
          <div
            className="radio-group"
          >
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_funeral_reimbursement_0_0"
                name="applicant_funeral_reimbursement_0"
                onChange={[Function]}
                type="radio"
                value="Yes"
              />
              <label
                className=""
                htmlFor="applicant_funeral_reimbursement_0_0"
              >
                Yes
              </label>
            </div>
            <div
              className="usa-radio"
            >
              <input
                checked={false}
                className=""
                id="applicant_funeral_reimbursement_0_1"
                name="applicant_funeral_reimbursement_0"
                onChange={[Function]}
                type="radio"
                value="No"
              />
              <label
                className=""
                htmlFor="applicant_funeral_reimbursement_0_1"
              >
                No
              </label>
            </div>
          </div>
        </fieldset>
      </div>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Continue
      </button>
    </div>
  </div>,
]
`;
